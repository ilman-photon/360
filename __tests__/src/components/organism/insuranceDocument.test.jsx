import { fireEvent, render } from "@testing-library/react";
import "@testing-library/jest-dom";
import InsuranceDocument from "../../../../src/components/organisms/InsuranceInformation/insuranceInformationNew";

window.scrollTo = jest.fn()

describe("SecurityQuestion Components", () => {
  let container;
  beforeEach(() => {
    container = render(<InsuranceDocument 
    />);
  });

  it("SecurityQuestionrender", () => {
    expect(container).toMatchSnapshot();
  });

  it("Set value to all field", () => {
    setTimeout(() => {
    const insuranceProvider = container.getByLabelText("Insurance Provider")
    fireEvent.change(insuranceProvider, { target: { value: "Insurance value" } });
    expect(insuranceProvider.value).toEqual("Insurance value");

    const planName = container.getByLabelText("Plan Name")
    fireEvent.change(planName, { target: { value: "Plan Name value" } });
    expect(planName.value).toEqual("Plan Name value");

    const subscriberMember = container.getByLabelText("Insurance Subscriber ID/ Member ID")
    fireEvent.change(subscriberMember, { target: { value: "Insurance Subscriber ID" } });
    expect(subscriberMember.value).toEqual("Insurance Subscriber ID");

    const groupId = container.getByLabelText("Group #")
    fireEvent.change(groupId, { target: { value: "Group Id" } });
    expect(groupId.value).toEqual("Group Id");

    const policyHolder = container.getByText("Policy Holder");
    expect("Policy Holder").toEqual(policyHolder.textContent);

    const isSubscriber = container.getByLabelText("No");
    fireEvent.click(isSubscriber);
    expect(isSubscriber.value).toEqual("no");

    const subscriberDetail = container.getByText("Enter the subscriber’s details");
    expect("Enter the subscriber’s details").toEqual(subscriberDetail.textContent);

    const firstName = container.getByLabelText("Subscriber First Name")
    fireEvent.change(firstName, { target: { value: "Test Name" } });
    expect(firstName.value).toEqual("Test Name");

    const lastName = container.getByLabelText("Subscriber Last Name")
    fireEvent.change(lastName, { target: { value: "Test Name" } });
    expect(lastName.value).toEqual("Test Name");

    const relationship = container.getByLabelText("Relationship")
    fireEvent.change(relationship, { target: { value: "Relationship value" } });
    expect(relationship.value).toEqual("Relationship value");

    const uploadText = container.getByText("Upload images of your insurance.");
    expect("Upload images of your insurance.").toEqual(uploadText.textContent);

    const insurancePriority = container.getByLabelText("Primary");
    fireEvent.click(insurancePriority);
    expect(insurancePriority.value).toEqual("primary");
}, 500);

  });

  it("Click Save Button", () => {
    setTimeout(() => {
      const button = container.getByText("Save")
      fireEvent.click(button);

      expect("Remove Insurance").toBeVisible()
    }, 500);
  });

  it("Click Update Button", () => {
    setTimeout(() => {
        const button = container.getByText("Edit")
        fireEvent.click(button);

    const insuranceProvider = container.getByLabelText("Insurance Provider")
    expect(insuranceProvider.value).toEqual("Insurance value");
    }, 500);
  });

  it("Render Fields for Update", () => {

    setTimeout(() => {
    const answer1 = container.getByLabelText(/Insurance Provider/i)
    fireEvent.change(answer1, { target: { value: "" } });
    expect(answer1.value).toEqual("");


        const button = container.getByText("Save")
      fireEvent.click(button);


      const postMessage = container.getByLabelText("This field is required")
      expect(postMessage).toBeTruthy()

    }, 500);
  });

});
