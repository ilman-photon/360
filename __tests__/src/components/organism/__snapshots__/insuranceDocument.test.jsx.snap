// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SecurityQuestion Components SecurityQuestionrender 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root css-8u5xrh"
      >
        <div
          class="MuiPaper-root MuiPaper-outlined MuiPaper-rounded MuiCard-root card css-1v5z4dq-MuiPaper-root-MuiCard-root"
        >
          <div
            class="MuiCardContent-root css-vp7ejv-MuiCardContent-root"
          >
            <div
              class="css-1p5q5e5-MuiStack-root"
            >
              <span
                class="MuiTypography-root MuiTypography-bodyLarge css-37jdci-MuiTypography-root"
              >
                You have no insurance on file
              </span>
            </div>
          </div>
        </div>
        <div
          style="position: fixed; bottom: 16px; right: 16px;"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root css-8u5xrh"
    >
      <div
        class="MuiPaper-root MuiPaper-outlined MuiPaper-rounded MuiCard-root card css-1v5z4dq-MuiPaper-root-MuiCard-root"
      >
        <div
          class="MuiCardContent-root css-vp7ejv-MuiCardContent-root"
        >
          <div
            class="css-1p5q5e5-MuiStack-root"
          >
            <span
              class="MuiTypography-root MuiTypography-bodyLarge css-37jdci-MuiTypography-root"
            >
              You have no insurance on file
            </span>
          </div>
        </div>
      </div>
      <div
        style="position: fixed; bottom: 16px; right: 16px;"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
